### ISSUE TITLE
not replicating

### ISSUE THREAD

**Issue by dreaganluna** [valenssoftware/bamboo-templatedplans-plugin#13]
I followed the steps in the readme, but it did not start replicating it's build plan jobs.
I also have no "template replication operations" menu item, as is shown in the image at the bottom of this page: https://iuliusathome.blogspot.be/p/template-build-and-deployment-for-bamboo.html.

I applied this to 2 already existing build plans, one was cloned from the other.
I also had to go into "tasks" of the second build plan before I could get to the correct misc-tab where the template and its sync options were listed. The main misc-tab just has the "is template" checkbox.

**Comment by giulio1979** [comment]
Hi,

let me try to explain it step by step:

- pick a plan which you would like to be a template, tick the "Is Template" flag. This will allow the plan to appear in the other plans in the dropdown with template names.

- Once you have the Template allowed, go in the configuration of the other plan - Miscellaneous page.
Here you should find a section "Template list" with a dropdown. Select from the dropdown the template name and tick what would you like to be templated.

The replication can be started in few modes (result is same but depending on your permissions in Bamboo one can be more comfortable than the other): 
- by doing changes to the template (save on Miscellaneous page)
- by going to the template operations menu
- from System administration / Addons / replication operations

As a test I would suggest to trigger the replication without any server pause. If you wait a bit you should see the progress in the replication operations.

J

**Comment by dreaganluna** [comment]
I triggered the replication process manually via the last bullet point you described and the template was copied over to the other build plan. Thanks!

I did try to change something small and save my changes to see if replication would pick that up. But that doesn't work. When I created a new task, I had to trigger the replication manually for it to be copied over to the other build plans.
When I changed something in an existing task, it wouldn't copy over, not even when I triggered the replication manually.

**Comment by giulio1979** [comment]
Hi again,

great to hear some things work, lets see some more hints:

- changes to the templates should be replicated but only after you trigger a replication manually, not each time you save. The replication is quite intrusive and a permanent / continuous replication may cause issues as build may be triggered with empty configuration exactly at the point when plan is being emptied and prepared for template copy.

- in order for templating to work properly please make sure you give your tasks unique names, this will ensure the engine can identify which task from template has been copied over and also will ensure the state of the tasks is maintained between copies.

J

**Comment by giulio1979** [comment]
Hi,

if you dont mind I will make a post on the blog with the info in this issue, seems we managed to pinpoint some difficulties and maybe we can make life easier for everyone.

J
